lib/components/sprite_stack_component.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:ui';
+
+import '../sprite.dart';
+import '../spritesheet.dart';
+import 'component.dart';
+
+class SpriteStackComponent extends PositionComponent {
+  SpriteSheet spriteSheet;
+
+  static final _sprites = <String, Map<double, Sprite>>{};
+
+  @override
+  set angle(double value) {
+    if (_sprites[spriteSheet.imageName][value] == null) {
+      _generated = false;
+    }
+    super.angle = value;
+  }
+
+  SpriteStackComponent(
+      {String imageName,
+      int textureWidth,
+      int textureHeight,
+      int columns,
+      int rows}) {
+    if (_sprites[imageName] == null) {
+      _sprites[imageName] = new Map();
+    }
+    x = 100;
+    y = 100;
+    width = textureWidth.toDouble();
+    height = textureHeight.toDouble();
+    spriteSheet = SpriteSheet(
+        imageName: imageName,
+        textureWidth: textureWidth,
+        textureHeight: textureHeight,
+        columns: columns,
+        rows: rows);
+  }
+
+  @override
+  void render(Canvas canvas) {
+    if (!loaded()) {
+      return;
+    }
+    prepareCanvas(canvas);
+    _sprites[spriteSheet.imageName][angle]
+        .render(canvas, width: width, height: height);
+  }
+
+  bool _generated = false;
+  bool _generating = false;
+
+  @override
+  bool loaded() {
+    for (var row = 0; row < spriteSheet.rows; row++) {
+      for (var column = 0; column < spriteSheet.columns; column++) {
+        var sprite = spriteSheet.getSprite(row, column);
+        if (!sprite.loaded()) {
+          return false;
+        }
+      }
+    }
+    generate();
+    return _generated;
+  }
+
+  void generate() async {
+    if (_generated || _generating) {
+      return;
+    }
+
+    _generating = true;
+    var pictureRecorder = PictureRecorder();
+    var canvas =
+        Canvas(pictureRecorder, Rect.fromLTWH(0.0, 0.0, width, height));
+
+    double index = 0;
+    canvas.rotate(-angle);
+    for (var row = 0; row < spriteSheet.rows; row++) {
+      for (var column = 0; column < spriteSheet.columns; column++) {
+        var spriteComponent = SpriteComponent.fromSprite(
+            spriteSheet.textureWidth.toDouble(),
+            spriteSheet.textureHeight.toDouble(),
+            spriteSheet.getSprite(row, column));
+        canvas.save();
+        spriteComponent.debugMode = false;
+        spriteComponent.angle = angle;
+        spriteComponent.y -= index++ - 5;
+        spriteComponent.render(canvas);
+        canvas.restore();
+      }
+    }
+
+    var pic = pictureRecorder.endRecording();
```

Comment:
since you don't re assigns these variables, you can make them final

Reply:
Ah yes, my IDE was already complaining, consider it done